featurebyte/service/preview.py

```diff
@@ -334,10 +363,15 @@ async def sample(
             feature_store_id=sample.feature_store_id,
         )
         if size > 0:
-            total_num_rows = await self._get_row_count(
-                session,
+            graph, node_name = self._get_row_count_graph_and_node(
                 graph=sample.graph,
                 node_name=sample.node_name,
+                sample_on_primary_table=sample_on_primary_table,
+            )
+            total_num_rows = await self._get_row_count(
```

Comment:
Would it make sense to pass the `sample_on_primary_table parameter` to `_get_row_count`, so it retrieves the correct graph and node for row count? This helps ensure correct usage and avoids introducing another `graph` variable here, which could be confusing alongside `sample.graph`.